tracks/dart/exercises/scrabble-score/mentoring.md

```diff
@@ -0,0 +1,56 @@
+# Mentoring
+
+## Reasonable solutions
+
+This exercise can be solved in different ways. We will use two of them. 
+Show both of them to get them, to get them to know more dart concepts.
+Map class will be used for both solutions.
+
+### Solution 1: using fold method ***_recommended_***
+```dart
+int score([String word]) {
+  var values = {
+    'A': 1, 'B': 3, 'C': 3, 'D': 2,
+    'E': 1, 'F': 4, 'G': 2, 'H': 4,
+    'I': 1, 'J': 8, 'K': 5, 'L': 1,
+    'M': 3, 'N': 1, 'O': 1, 'P': 3,
+    'Q': 10, 'R': 1, 'S': 1, 'T': 1,
+    'U': 1, 'V': 4, 'W': 4, 'X': 8,
+    'Y': 4, 'Z': 10,
+  };
+
+  return word.toUpperCase().split("").fold(0, (prev, curr) => prev + values[curr]);
+}
+```
+
+### Solution 2: using map and reduce methods
+```dart
+int score([String word]) {
+  var values = {
+    'A': 1, 'B': 3, 'C': 3, 'D': 2,
+    'E': 1, 'F': 4, 'G': 2, 'H': 4,
+    'I': 1, 'J': 8, 'K': 5, 'L': 1,
+    'M': 3, 'N': 1, 'O': 1, 'P': 3,
+    'Q': 10, 'R': 1, 'S': 1, 'T': 1,
+    'U': 1, 'V': 4, 'W': 4, 'X': 8,
+    'Y': 4, 'Z': 10,
+  };
+
+  return word.toUpperCase().split("").map((e) => values[e.toString()]).reduce((prev,curr) => prev + curr);
+}
+```
+
+## Common suggestions
+- Students need to know how [Map][reference-map-class] works, because they need it for both of the solutions
```

Comment:
````suggestion
## Common suggestions

- Students need to know how [Map][reference-map-class] works, because they need it for both of the solutions
````